src/FirstSteps/RPG/Heroes/Hero.cs

```diff
@@ -16,7 +17,8 @@ public abstract class Hero : IAttacker
         protected int _intelligence;
         protected int _agility;
         protected int _health;
-        public int Health { get { return _health; } set { } }
+        public int _storedHealth { get; set; }
```

Comment:
1. First, you need to store maximum health for internal usage, so you need the `protected int _maxHealth;` variable.
2. Then, you have to think where you are setting the initial value of the `_health` variable for Elf/Human/Undead/Dwarf and set exactly the same value in the same place for `_maxHealth`.

_Extra task: think if it would be possible to set `_maxHealth` as a read-only variable because it does not have to change after initialization._